rasa/core/policies/ted_policy.py

```diff
@@ -375,12 +393,33 @@ def _create_label_data(
     ) -> Tuple[RasaModelData, List[Dict[Text, List["Features"]]]]:
         # encode all label_ids with policies' featurizer
         state_featurizer = self.featurizer.state_featurizer
-        encoded_all_labels = state_featurizer.encode_all_actions(domain, interpreter)
+        encoded_all_labels = state_featurizer.encode_all_labels(domain, interpreter)
 
         attribute_data, _ = convert_to_data_format(
             encoded_all_labels, featurizers=self.config[FEATURIZERS]
         )
 
+        label_data = self._assemble_label_data(attribute_data, domain)
+
+        return label_data, encoded_all_labels
+
+    def _assemble_label_data(
+        self, attribute_data: Data, domain: Domain
+    ) -> RasaModelData:
+        """Construct data regarding labels to be fed to the model.
+
+        The resultant model data should contain the keys
+        `label_action_name`, `label_action_text` and `label`.
+        `label_action_*` will contain the sequence, sentence and mask features
+        for corresponding labels and `label` will contain the numerical label ids.
+
+        Args:
+            attribute_data: Feature data for all labels.
+            domain: Domain of the assistant.
+
+        Returns:
+            Features of labels ready to be fed to the model.
+        """
         label_data = RasaModelData()
         label_data.add_data(attribute_data, key_prefix=f"{LABEL_KEY}_")
         label_data.add_lengths(
```

Comment:
I'm a bit confused about what's going on here, the docstring says it should contain `label_action_name` but I only see `label_action_text` being added?

Reply:
`attribute_data` will contain `label_action_name` and `label_action_text`. Hence, when `add_data` is called, they should be present in `label_data`.

Reply:
Thank you!